### PR TITLE
Fix YUNMAI muscle value

### PR DESCRIPTION
**Pull request by d3m3vilurr**
this value doesn't be related `sex` only related `body_type`
in the original app, if body_type is 1(fitness body type),it should multiply
with 0.7.
if not (normal body type), it should multiply with 0.67

### CODE REVIEW

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/lib/YunmaiLib.java [hunk 1]
```diff
@@ -51,10 +51,6 @@ public float getMuscle(float bodyFat) {
         float muscle;
         muscle = (100.0f - bodyFat) * 0.67f;
 
-        if (sex == 1) {
```
Comment (d3m3vilurr) [inline android_app/app/src/main/java/com/health/openscale/core/bluetooth/lib/YunmaiLib.java]: IMO, good option is condition change to `if (activityLevel == 5) {`
@oliexdev can you suggest it?

Comment (oliexdev) [reply android_app/app/src/main/java/com/health/openscale/core/bluetooth/lib/YunmaiLib.java]: I am not sure, it was a long time ago as I reverse engineered it. I don't own anymore this scale, so I can't test it but this error could be related to issue #484 
So does the muscle values are correct if you change it to activityLevel == 5?

**Comment by d3m3vilurr** [comment]
well... i reversed for several patches include this and #522. for example. at least, this patch will return same value as original app.
(but I don't know `YUNMAI-ISM2-W` case & I also need dump packet info.

----
and these are background info of the question.
IMO, yunmai try to make the newer version devices to support athletic people. and in this case, device version should be over the `0x1E`; and this project only cares `0x1E` version devices. (and now, app can support 0x13 ver ;))

for example, these are specialized part of the functions
1. getFat

<details><summary>original</summary>

```
  if (((iVar2 == 0) && (0x1e < device_version)) && (body_type == 1)) {
    fVar7 = getBMI(weight,height);
    fVar3 = (float)((uint)weight & 0xffffff00 | (uint)(byte)(unkfloat1)fVar7) - 22.50000000;
    fVar7 = ROUND(fVar3);
    if ((int)fVar7 < 0) {
      fVar7 = (float)-(int)fVar7;
    }
    uVar1 = SUB84((double)(fVar3 * (float)(longlong)(int)ROUND(fVar3 / (float)(longlong)(int)fVar7))
                  ,0);
    pow(__x,__y);
    if (sex == 1) {
      fVar4 = 12.00000000;
    }
    else {
      if (sex != 2) {
LAB_00011596:
        fVar6 = 0.00000000;
        goto LAB_00011600;
      }
      fVar4 = 22.79999924;
    }
    Var5 = SUB41((fVar4 - fVar6) +
                 (float)((double)CONCAT44(extraout_r1_00,uVar1) *
                        (double)(longlong)(int)ROUND(fVar3 / (float)(longlong)(int)fVar7)),0);
  }
```

</details>

```
            float a = bmi - 22.5f;
            float b = Math.abs(a);
            float c = a * a / b;
            float d = (float)Math.pow(c, c);

            if (this.sex == 1) {
                fat = (12.0f - r) + (d * a / b);
            } else {
                fat = (22.8f - r) + (d * a / b);
            }
```

2. getMuscle

<details><summary>original</summary>

```
  if ((iVar1 == 0) && (body_type == 1)) {
    fVar2 = (100.00000000 - fat) * 0.69999999;
  }
```

</details>

```
        if (this.fitnessBodyType) {
            muscle = (100.0f - bodyFat) * 0.7f;
        }

```

3. getVisceralFat

<details><summary>original</summary>

```
// reversed java method
            float vf;
            if (bodyFat > 15.0f) {
                vf = (bodyFat - 15.0f) / 1.1f + 6.0f;
            } else {
                vf = -1 * (15.0f - bodyFat) / 1.4f + 6.0f;
            }
            float vf = vf + 6;
            if (vf < 1.0f) {
                return 1.0f;
            }
            if (vf > 9.0f) {
                return 9.0f;
            }
            return vf;
```

</details>

```
            float vf;
            if (bodyFat > 15.0f) {
                vf = (bodyFat - 15.0f) / 1.1f + 6.0f;
            } else {
                vf = -1 * (15.0f - bodyFat) / 1.4f + 6.0f;
            }
            float vf = vf + 6;
            if (vf < 1.0f) {
                return 1.0f;
            }
            if (vf > 9.0f) {
                return 9.0f;
            }
            return vf;

```

4. getSkeletalMuscle

<details><summary>original</summary>

```
// I found it in the 2.5 version libconfig.so
// & quite similar as getMuscle. only diff the magic numbers
  fVar2 = (100.00000000 - fat) * 0.52999997;
  iVar1 = strcmp(__s1,"en");
  if ((iVar1 == 0) && (body_type == 1)) {
    fVar2 = (100.00000000 - fat) * 0.60000002;
  }
  return (float)SUB41((float)(longlong)SUB84(ROUND((double)(fVar2 * 100.00000000) + 0.50000000),0) /
                      100.00000000,0);
```

</details>


```
    public float getSkeletalMuscle(float bodyFat) {
        float muscle;

        muscle = (100.0f - bodyFat) * 0.53f;
        if (this.fitnessBodyType) {
            muscle = (100.0f - bodyFat) * 0.6f;
        }

        muscle = (muscle * 100.0f) + 0.5f / 100.0f;

        return muscle;
    }
```


sadly I don't have these newer device, also I couldn't find a setting option to change the body type in original app.
(so, I couldn't test fitness body type in original app.)
but basically every person would be normal type, that we set `0x03` in `user_add_or_query`


I'll reverse `user_add_or_query` part to check to send diff body type instead `0x03`.

**Comment by oliexdev** [comment]
close this PR because its implemented in PR #524

**Pull request closed without merging**